File: scanner-fixtures/src/real_llm_calls.ts
/**
 * Example LLM call sites for scanner validation — one OpenAI, one Anthropic.
 * Model IDs come from environment variables only (no literals).
 */

import OpenAI from "openai";
import OpenAI from "openai";

export async function openaiChatCompletion() {
  const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
  return client.chat.completions.create({
    model: process.env.OPENAI_MODEL,
    messages: [
      { role: "system", content: "You answer briefly." },
      { role: "user", content: "What is a circuit breaker?" },
    ],
  });
}

export async function anthropicMessagesCreate2() {
  const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
  return client.chat.completions.create({
    model: "gpt-5.4-mini",
    max_tokens: 512,
    messages: [{ role: "user", content: "Whats the weather in Tokyo?" }],
  });
}

File: scanner-fixtures/src/real_llm_calls.py
"""
Example LLM call sites for scanner validation — one OpenAI, one Anthropic.
Model IDs come from environment variables only (no literals).
"""

from __future__ import annotations

import os
from typing import Any

from google.generativeai import GenerativeModel

GOOGLE_API_KEY = os.getenv("GOOGLE_API_KEY")
google_model='gemini-2.0-flash'

def _env(name: str) -> str | None:
    return os.getenv(name)


def openai_chat_completion() -> Any:
    client = OpenAI(api_key=_env("OPENAI_API_KEY"))
    return client.chat.completions.create(
        model=OPENAI_API_KEY,
        messages=[
            {"role": "system", "content": "You answer briefly."},
            {"role": "user", "content": "What is a circuit breaker?"},
        ],
    )


def anthropic_messages_create() -> Any:
def anthropic_messages_create() -> Any:
    client = GenerativeModel(api_key=_env("GOOGLE_API_KEY"), model_name=google_model)
    return client.generate_content(
        contents="What is a circuit breaker?"
    )

